### PR TITLE
optimize cdc log file

### LINKED ISSUE #4450: Update the default value of log-file for cdc
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

**What did you do?**

**What did you expect to see?**
Currently, we set `/dev/stderr` as the default value for the log-file of cdc, however, it seems that TiCDC takes it as a regular file and write to the root dir of the container:
middle_img_v2_9fffac76-e1c6-4839-bd1e-e2d7a9b020dg
We need to set the default value to "".
**What did you see instead?**

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
Fix #4450.

### What is changed and how does it work?
Ref doc https://docs.pingcap.com/zh/tidb/stable/deploy-ticdc , use `cdc.log` filename as default value.
Fix tolerations type error.
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: test.log [hunk 1]
```diff
@@ -0,0 +1,817 @@
+level=debug ts=2022-03-13T15:41:46.2397191Z caller=writer.go:94 component=receive component=receive-writer msg="Out of bounds metric" lset="{__name__=\"tikv_config_raftstore\", cluster=\"basic\", component=\"tikv\", instance=\"basic-tikv-0\", job=\"default-basic-tikv\", kubernetes_namespace=\"default\", name=\"lock_cf_compact_bytes_threshold\", prometheus_replica=\"default_basic-monitor-0\", tidb_cluster=\"default-basic\"}" sample="unsupported value type"
```
Comment (csuzhangxc) [inline test.log:1]: you should not commit this file

File: examples/ticdc/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# A Basic TiDB cluster with monitoring
+
+> **Note:**
+>
+> This setup is for test or demo purpose only and **IS NOT** applicable for critical environment. Refer to the [Documents](https://pingcap.com/docs/stable/tidb-in-kubernetes/deploy/prerequisites/) for production setup.
+
+The following steps will create a TiDB cluster with monitoring, the monitoring data is not persisted by default.
+
+**Prerequisites**: 
+- Has TiDB operator `v1.1.0-beta.1` or higher version installed. [Doc](https://pingcap.com/docs/stable/tidb-in-kubernetes/deploy/tidb-operator/)
```
Comment (csuzhangxc) [inline examples/ticdc/README.md:10]: this is not correct for this TiCDC example?

File: pkg/apis/pingcap/v1alpha1/tidbcluster.go [hunk 3]
```diff
@@ -913,7 +913,7 @@ func (tc *TidbCluster) TiCDCLogFile() string {
 		}
 	}
 
-	return "/dev/stderr"
+	return "cdc.log"
```
Comment (csuzhangxc) [inline pkg/apis/pingcap/v1alpha1/tidbcluster.go:916]: How about setting to `""` as suggested in the issue's description? `We need to set the default value to "".`

File: examples/advanced/tidb-cluster.yaml [hunk 4]
```diff
@@ -866,7 +866,7 @@ spec:
   #     timezone: UTC
   #     gcTTL: 86400
   #     logLevel: info
-  #     logFile: /dev/stderr
+  #     logFile: cdc.log
```
Comment (DanielZhangQD) [inline examples/advanced/tidb-cluster.yaml:869]: We cannot log to file, we have to log it to stdout.

File: examples/ticdc/README.md (hunk 2 above)
Comment (DanielZhangQD) [reply examples/ticdc/README.md:10]: @mikechengwei I think no need to add an example for TiCDC, it has been described clearly in the docs and the advanced example.

**Comment by DanielZhangQD** [comment]
@mikechengwei Have you reproduced this issue in your cluster?
Actually, we have not been able to reproduce it in our cluster yet.

**Comment by mikechengwei** [comment]
> @mikechengwei Have you reproduced this issue in your cluster? Actually, we have not been able to reproduce it in our cluster yet.

I will reproduce it

**Pull request closed without merging**